### PR TITLE
Fix two leaks in the Mac platform code.

### PR DESCRIPTION
**Pull request by pcwalton**
Related: servo/servo#10029

r? @metajack

### CODE REVIEW

File: platform/macos/mod.rs [hunk 1]
```diff
@@ -591,6 +591,8 @@ fn select(port: mach_port_t, blocking_mode: BlockingMode)
                         os_result => return Err(MachError(os_result)),
```
Comment (metajack) [inline platform/macos/mod.rs]: Can we leak here in the case that we get a bad os_result?

**Comment by metajack** [comment]
Looks good aside from the one comment.

**Comment by pcwalton** [comment]
Fixed. r? @metajack

**Pull request merged**